### DISCUSSION TITLE
Reuse transaction

### DISCUSSION THREAD

**Discussion by auxsvr** [sqlalchemy/sqlalchemy#8610] [Usage Questions] [answered]
Writing an `upsert` function for Postgresql, I would like to use the enclosing transaction instead of starting a new one. It seems that the session does not allow this. Any suggestion?

**Comment by zzzeek** [comment, marked as answer]
hi there -

there is no limitation on using a particular transaction for multiple operations, with the exception on PostgreSQL of some specific DDL operations like creating schemas, which an "upsert" is not.    would need specifics to understand the problem you are having.

**Reply by auxsvr** [reply]
Hi,
You're right, I was using the session as a context manager, which closes on exit. I'm sorry for wasting your time.
